src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -2669,3 +2858,739 @@ def costing_upper_bound_eq(c, p):
+
+        For escalating costs, the series present worth factor is modified to
+        account for escalation, yielding a modified formula
+
+        P/A(r, g, N) = ( 1 - [ (1+g)**(N) ] * [(1+r)**(-N)] ) / (r - g)
+
+        where r is the discount rate expressed as a decimal, N is the project
+        lifetime, and g is the escalation rate (e.g. inflation) expressed as a
+        decimal.
+
+        The general formulation considers a capital escalation period followed
+        by the operating period, and capital costs may be distributed across
+        the capital escalation period rather than fully paid for upfront. For
+        example, if the capital expenditures are distributed across a 3-year
+        capital escalation period, the PV from the capital costs are given as
+
+        PV_Capital_Cost = Y1_% * CAPEX * P/A(r, gCap, 1)
+                          + Y2_% * CAPEX * P/A(r, gCap, 2)
+                          + Y3_% * CAPEX * P/A(r, gCap, 3)
+
+        where Y1_%, Y2_%, and Y3_% are the percentage of capital expenditures
+        in each year expressed as decimals, CAPEX is the total capital cost from
+        equipment purchasing, and gCap is the capital escalation growth rate
+        expressed as a decimal. The capital costs spent in each year are handled
+        separately to properly account for the value growth over time.
+
+        Revenue, operating costs and royalties based on revenue escalate with
+        standard inflation. Notably, these cash flows occur after any capital
+        expenditure period, meaning that the annuity growth must be offset by
+        the length of the capital expenditure period. This yields the expressions
+
+        PV_Revenue = REVENUE * [ P/A(r, gRev, NOp+NCap) - P/A(r, gRev, NCap) ]
+
+        PV_Operating_Cost = OPEX * [ P/A(r, gOp, NOp+NCap) - P/A(r, gOp, NCap) ]
+
+        PV_Royalties = iRoy_% * REVENUE * [ P/A(r, gRev, NOp+NCap) - P/A(r, gRev, NCap) ]
+
+        where REVENUE is the annual revenue, OPEX is the annual operating cost,
+        gRev is the inflation or growth rate of revenue year-on-year expressed as
+        a decimal, gOp is the inflation or growth rate of operating costs year-on-year
+        expressed as a decimal, NOp is the length of the operating period or plant
+        lifetime, NCap is the length of the capital expenditure period, and iRoy_%
+        is the percentage of the revenue charged as royalties expressed as a decimal.
+        The expressions above take the annuity growth during the entire analysis
+        period (NOp+NCap) and subtract the capital expenditure period (NCap) as there
+        is no operation or production during that time.
+
+        Finally, the full formulation supports estimation of interest
+        owed on loans and value losses due to capital depreciation. The loan
+        interest is calculated as an annual loan payment
+
+        Annual_Loan_Payment = Debt * [ iLoan_% * (1 + iLoan_%)**Nloan ] / [ (1 + iLoan_%)**Nloan - 1 ]
```

Comment:
I am a little bit confused about why the loan payment is included in the calculation. Typically, the interest is accounted for in the present worth factor. In the later calculations, it appears that you are already converting everything to present value. If this is the case, you do not need to include the loan interest in your summation for NPV (but please let me know if i am misinterpreting something in the formulation).

Reply:
I think you correct in your assessment. I understood that the interest should be included as a cost, but if it is already included in capital component then I can remove the loan repayment from the formulation. I'm happy to discuss it further here or offline.

Reply:
The case study source provides a few rates - 3% inflation for operating costs, 3% inflation for revenue, 10% discount rate, 3.6% escalation rate for capital costs, and 6% interest rate on loans. Would it correct to use the 6% interest rate as the discount rate when projecting capital expenditure costs, with a 3.6% growth rate? And then 10% (in this scenario) would be the discount rate for operating costs, revenue and royalties, with a 3% growth rate.